pkg/crc/machine/vsock.go

```diff
@@ -0,0 +1,58 @@
+package machine
+
+import (
+	"fmt"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/daemonclient"
+	"github.com/code-ready/gvisor-tap-vsock/pkg/types"
+	"github.com/pkg/errors"
+)
+
+func exposePorts() error {
+	daemonClient := daemonclient.New()
+	requestedPorts := vsockPorts()
+	alreadyOpenedPorts, err := daemonClient.List()
+	if err != nil {
+		return err
+	}
+	var missingPorts []types.ExposeRequest
+	for _, port := range requestedPorts {
+		if !isOpened(alreadyOpenedPorts, port) {
+			missingPorts = append(missingPorts, port)
+		}
+	}
+	for i := range missingPorts {
+		port := &requestedPorts[i]
+		if err := daemonClient.Expose(port); err != nil {
+			return errors.Wrapf(err, "failed to expose port %s -> %s", port.Local, port.Remote)
+		}
+	}
+	return nil
+}
+
+func isOpened(exposed []types.ExposeRequest, port types.ExposeRequest) bool {
+	for _, alreadyOpenedPort := range exposed {
+		if port == alreadyOpenedPort {
+			return true
+		}
+	}
+	return false
+}
+
+func vsockPorts() []types.ExposeRequest {
+	return []types.ExposeRequest{
+		{
+			Local:  fmt.Sprintf(":%d", constants.VsockSSHPort),
+			Remote: "192.168.127.2:22",
```

Comment:
can we name these variables?

`Remote: [remotehost]:[ssh]`

Reply:
done